pulsar-broker-auth-sasl/src/main/java/org/apache/pulsar/broker/authentication/AuthenticationProviderSasl.java

```diff
@@ -98,8 +102,14 @@ public void initialize(ServiceConfiguration config) throws IOException {
                 throw new IOException(e);
             }
         }
-
-        this.signer = new SaslRoleTokenSigner(Long.toString(new Random().nextLong()).getBytes());
+        String saslJaasServerRoleTokenSignerSecretFile = config.getSaslJaasServerRoleTokenSignerSecret();
+        byte[] secret = null;
+        if (StringUtils.isNotBlank(saslJaasServerRoleTokenSignerSecretFile)) {
+            secret = readSecretFromUrl(saslJaasServerRoleTokenSignerSecretFile);
+        } else {
+            secret = SaslConstants.JAAS_DEFAULT_ROLE_TOKEN_SIGNER_SECRET.getBytes();
```

Comment:
What is the purpose of defaulting to a static value? I think we should maintain the current default (using a random Long), and then users that want to use a shared token can do so. Otherwise, users might not realize they need to update the value and the signing secret would be easy to guess, assuming I understand correctly.

Reply:
if we should maintain the current default (using a random Long)，this will cause users to be unable to use under multiple brokers.

Reply:
I agree fail on startup if the value is null while authentication is enabled for SASL.